compiler/modules/front/codeGen.bal

```diff
@@ -495,12 +497,19 @@ function codeGenWhileStmt(CodeGenContext cx, bir:BasicBlock startBlock, Environm
     }
 }
 
-function validLoopAssignments(CodeGenContext cx, int[] assignments) returns CodeGenError? {
+function validLoopAssignments(CodeGenContext cx, Binding? binding, int[] assignments) returns CodeGenError? {
+    // pr-todo: I think swapping the loop order will be more efficient
     foreach int r in assignments {
-        if r < cx.loopStartRegister() {
-            return cx.semanticErr(`assignment to narrowed variable ${<string>cx.registerVarName(r)} in loop`, <Position>cx.registerPosition(r));
+        Binding? tem = binding;
+        while tem != () {
```

Comment:
This feels like we don't have the right data structure any more. Should we be keeping more information about each assignment?

Reply:
Yes, since we'll have to keep the position anyway, we should also keep narrowed binding to which the assignment happens (currently we keep unnarrowed binding's reg number). Then we'll have original logic here, just with narrowed reg number instead unnarrowed reg number.